coingecko/src/endpoint/price.ts

```diff
@@ -21,62 +21,104 @@ const presetTickers: { [ticker: string]: string } = {
+export const getPriceData = async (
+  config: Config,
+  ids: string,
+  currency: string,
+  marketCap = false,
+) => {
   const url = '/simple/price'
-  const market = validator.validated.data.quote
-  const coin = await convertFromTicker(config, symbol, coinid)
-
   const params = {
-    ids: coin,
-    vs_currencies: market,
+    ids,
+    vs_currencies: currency.toLowerCase(),
+    include_market_cap: marketCap,
   }
-
   const options = {
     ...config.api,
     url,
     params,
   }
+  const response = await Requester.request(options)
+  return response.data
+}
 
-  const response = await Requester.request(options, customError)
-  const result = Requester.validateResultNumber(response.data, [
-    coin.toLowerCase(),
-    market.toLowerCase(),
-  ])
+export const getIdtoSymbol = (symbols: string[], coinList: any) => {
+  return Object.fromEntries(
+    symbols.map((symbol) => {
+      if (symbol in presetTickers) {
+        return [presetTickers[symbol], symbol]
+      }
+      const coin = coinList.find(
+        (d: any) =>
+          d.symbol.toLowerCase() === symbol.toLowerCase() &&
+          !blacklist.includes(d.id.toLowerCase()),
+      )
+      return [coin.id, symbol]
+    }),
+  )
+}
+
+export interface Prices {
+  [symbol: string]: number
+}
+
+const getPayload = (symbols: string[], prices: Prices, quote: string) => {
+  const payloadEntries = symbols.map((symbol) => {
+    const key = symbol.toUpperCase()
+    const val = {
+      quote: {
+        [quote.toUpperCase()]: {
+          price: prices[symbol],
+        },
+      },
+    }
+    return [key, val]
+  })
 
+  const payload: ResponsePayload = Object.fromEntries(payloadEntries)
+  return payload
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, customParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const base = validator.validated.data.base
+  const symbols = Array.isArray(base) ? base : [base]
+  const coingeckoSymbolId = validator.validated.data.coinid
+  const quote = validator.validated.data.quote
+
+  const coinList = await getCoinList(config)
+  const idToSymbol = getIdtoSymbol(symbols, coinList)
+  const ids = coingeckoSymbolId
+    ? coingeckoSymbolId.toLowerCase()
+    : Object.keys(idToSymbol).join(',')
+
+  const response: Record<string, any> = await getPriceData(config, ids, quote)
+  const prices = Object.fromEntries(
+    Object.entries(response).map(([coinId, data]) => [
+      idToSymbol[coinId],
+      Requester.validateResultNumber(data, [quote.toLowerCase()]),
+    ]),
+  )
+
+  const result = symbols.length === 1 ? prices[symbols[0]] : ''
+  const payload = getPayload(symbols, prices, quote)
   return Requester.success(jobRunID, {
-    data: config.verbose ? { ...response.data, result } : { result },
+    data: config.verbose ? { ...response.data, result, payload } : { result, payload },
```

Comment:
I like the concept of having a different key for batched responses.

Reply:
For batched and for anything that can't be included into a single result